pkg/cluster/sync.go

```diff
@@ -715,6 +713,12 @@ func (c *Cluster) syncPreparedDatabases() error {
 		if err := c.syncExtensions(preparedDB.Extensions); err != nil {
 			return err
 		}
+
+		if err := c.closeDbConn(); err != nil {
+			c.logger.Errorf("could not close database connection: %v", err)
+		} else {
+			c.logger.Debugf("closed preparedDbName: %s", preparedDbName)
+		}
```

Comment:
I would remove the else block. Don't see much value logging this info.

Reply:
> Hey @neelasha-09. Cool that you found the issue. Sorry was in vacation. Can you reflect my comments and we'll merge it. Have to look, if I've used the defer within loops someone else.

We already checked and it's here: #1130

Reply:
Ah, thanks.